core/src/main/java/org/bitcoinj/core/StoredBlock.java

```diff
@@ -172,6 +179,22 @@ public static StoredBlock deserializeCompact(NetworkParameters params, ByteBuffe
         return new StoredBlock(params.getDefaultSerializer().makeBlock(header), chainWork, height);
     }
 
+    /**
+     * Deserializes the stored block from a custom packed format. Used internally.
+     *
+     * @param buffer data to deserialize
+     * @return deserialized stored block
+     */
+    public static StoredBlock deserializeCompactV2(NetworkParameters params, ByteBuffer buffer) throws ProtocolException {
+        byte[] chainWorkBytes = new byte[StoredBlock.CHAIN_WORK_BYTES_V2];
+        buffer.get(chainWorkBytes);
+        BigInteger chainWork = new BigInteger(1, chainWorkBytes);
+        int height = buffer.getInt();  // +4 bytes
```

Comment:
nit:
How do we know it's '4' bytes? The line below clearly is adding 1 more byte but this one is getting a parameter `ByteBuffer` that I guess there's no way to be really sure how many bytes it will have. So I would rather remove the `// +4 bytes` comment.